Enhance Sidebar Slot properties & fix a minor UI bug

tecoholic opened the pull request:
## Description

* This PR adds some contextual properties as slot properties to the `CourseAuthoringOutlineSidebarSlot` and the `CourseAuthoringUnitSidebarSlot`. This will allow the plugins in these slots to provide better contextual information based on the content that's being rendered in the Course Outline and Unit pages.
* Fixes a minor bug where a container on the sidebar's height wasn't set to 100% causing its child elements to be rendered in limited space. This was affecting the plugin-slot components and is being rectified in this PR.

Useful information to include:
- Which edX user roles will this change impact? "Course Author" and
"Developer".


## Supporting information

Improves upon the implementation from https://github.com/openedx/frontend-app-authoring/pull/1752

## Testing instructions

Please provide detailed step-by-step instructions for testing this change.


## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.

farhaanbukhsh reviewed (approved):
👍 

- ✅ I tested this on the sandbox and it works smoothly.
- ✅  I read through the code
- ❌  I checked for accessibility issues
- ❌  Includes documentation

farhaanbukhsh commented:
@brian-smith-tcril Can you please review this when you have sometime and merge it if it's okay. :)

brian-smith-tcril commented on src/course-unit/CourseUnit.jsx:
```diff
@@ -220,12 +220,13 @@ const CourseUnit = ({ courseId }) => {
             </Layout.Element>
             <Layout.Element>
-              <Stack gap={3}>
+              <Stack gap={3} className="h-100">
```
Could you please make this change a separate PR? That should be a very quick approval/merge.

tecoholic replied:
@brian-smith-tcril Sure thing! Here it is. https://github.com/openedx/frontend-app-authoring/pull/1852

tecoholic replied:
@brian-smith-tcril Sure thing! Here it is. https://github.com/openedx/frontend-app-authoring/pull/1852

brian-smith-tcril commented on src/plugin-slots/CourseAuthoringUnitSidebarSlot/index.tsx:
```diff
@@ -10,11 +10,14 @@ export const CourseAuthoringUnitSidebarSlot = (
     blockId,
     courseId,
     unitTitle,
+    verticalBlocks,
```
I know the term `vertical` has been used internally for a long time, but I'm not sure if it's the public-facing name for this. By making it part of `pluginProps` we're "committing" to that as an "official" name by making it part of the explicit API.

I don't know if there's a name we'd prefer here. @arbrandes thoughts?

farhaanbukhsh replied:
@brian-smith-tcril @arbrandes we need these fixes for incontext-metric so it will be great if we can wind this up before teak cut.

arbrandes replied:
There's always been a dichotomy with the naming of the "unit" concept.  In the OLX itself it's "vertical", so we'll likely never be rid of the name.  However, since the prop immediately above it is `unitTitle`, it would make sense to align both to use the same terminology.

tecoholic replied:
@farhaanbukhsh @arbrandes I just realized something. These are actually verticalBlock's children. So these would be just `xblocks`. Correcting it now.

bradenmacdonald replied:
I think it means "vertical blocks" as in "the blocks in the vertical".

> In the OLX itself it's "vertical", so we'll likely never be rid of the name.

In learning core and libraries, we are moving to use the term "unit" everywhere in the code and UI, except for OLX.

brian-smith-tcril reviewed (commented):
Overall this makes sense and is looking good. I left a few small comments, and I have one other request:

Could you make sure to update the documentation for these slots to include the new props? Also, adding some examples that show the props being used would be wonderful, but not a blocker.

tecoholic commented:
@brian-smith-tcril Thanks for reviewing this. I have updated the docs and added an extra example for the Unit Page sidebar with a screenshot. I couldn't come up with a quick filter example for the course outline one.
